framework/src/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -51,29 +79,60 @@ public Void visitAnnotation(AnnotationTree node, Void p) {
         }
 
         Element element = TreeInfo.symbol((JCTree) node.getAnnotationType());
-        if (!(element.toString().equals(ArrayLen.class.getName())
+
+        if (element.toString().equals(IntRange.class.getName())) {
+            if (args.size() == 2
```

Comment:
That's right.  When a programmer sees a test, the programmer assumes that the case can occur.

If something cannot occur, then please throw an exception instead, such as

```java
if (!(args.size() == 2
      && args.get(0).getKind() == Kind.ASSIGNMENT
      && args.get(1).getKind() == Kind.ASSIGNMENT)) {
  throw new Exception("This can't happen");
}
```

Another (minor) advantage of this is that it doesn't add an extra level of indentation to the main code.